imageroot/bin/discover-service

```diff
@@ -14,12 +14,12 @@ import sys
 import json
 import agent
 
-mail_server_to_use = os.environ["MAIL_SERVER_TO_USE"]
+mail_server = os.environ["MAIL_SERVER"]
 rdb = agent.redis_connect(host="127.0.0.1") # full read-only access on every key
 
 # we query about TCP service provided by the mail server
-smtp_port = rdb.hget(f"module/{mail_server_to_use}/srv/tcp/submission", "port") or ""
-imap_port = rdb.hget(f"module/{mail_server_to_use}/srv/tcp/imap", "port") or ""
+smtp_port = rdb.hget(f"module/{mail_server}/srv/tcp/submission", "port") or ""
+imap_port = rdb.hget(f"module/{mail_server}/srv/tcp/imap", "port") or ""
 
 agent.set_env("ROUNDCUBEMAIL_DEFAULT_PORT", imap_port)
```

Comment:
agent.set_env does not work here: it is not an action step!

You should implement something like https://github.com/NethServer/ns8-kickstart/pull/19

- write a file with relevant vars
- load them in the .service unit (or with podman run --env-file)

Reply:
done